x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx

```diff
@@ -0,0 +1,323 @@
+
+      const dataViewId = getDataViewByIndexPattern(index)?.id;
+      if (dataViewId) {
+        const sloDataView = await dataViewsService.get(dataViewId);
+        setDataView(sloDataView);
+        getQuery();
+      }
+    };
+
+    const getQuery = () => {
+      const esSearchRequest = getESQueryForLogRateAnalysis(
+        params,
+        groupBy,
+        groupings
+      ) as QueryDslQueryContainer;
+      if (esSearchRequest) {
+        setEsSearchQuery(esSearchRequest);
+      }
+    };
+    getDataView();
+  }, [index, dataViews, params, dataViewsService, groupBy, groupings]);
+
+  // Identify `intervalFactor` to adjust time ranges based on alert settings.
+  // The default time ranges for `initialAnalysisStart` are suitable for a `1m` lookback.
+  // If an alert would have a `5m` lookback, this would result in a factor of `5`.
+  // If an alert is just starting, the visible time range to look back might not cover the
+  // long window yet, only then we'll take the short window to look back,
+  // otherwise it will be the long window.
+  const alertActionGroup = getActionGroupFromReason(alert.reason);
+  const relatedWindow = (
+    (alert.fields[ALERT_RULE_PARAMETERS]?.windows ?? []) as WindowSchema[]
+  ).find((window: WindowSchema) => window.actionGroup === alertActionGroup);
+
+  const longWindowValue = relatedWindow?.longWindow.value;
+  const longWindowUnit = relatedWindow?.longWindow.unit;
+  const longWindowLookbackDuration =
+    longWindowValue && longWindowUnit
+      ? moment.duration(longWindowValue as number, longWindowUnit as any)
+      : moment.duration(1, 'm');
+  const longWindowLookbackDurationAsSeconds = longWindowLookbackDuration.asSeconds();
+
+  const shortWindowValue = relatedWindow?.shortWindow.value;
+  const shortWindowUnit = relatedWindow?.shortWindow.unit;
+  const shortWindowLookbackDuration =
+    shortWindowValue && shortWindowUnit
+      ? moment.duration(shortWindowValue as number, shortWindowUnit as any)
+      : moment.duration(1, 'm');
+  const shortWindowLookbackDurationAsSeconds = shortWindowLookbackDuration.asSeconds();
+
+  // @ts-ignore
+  const dataTimeRange = getDataTimeRange(alert.fields[ALERT_TIME_RANGE], relatedWindow);
+  const timeRange = { min: moment(dataTimeRange.from), max: moment(dataTimeRange.to) };
+  const alertStart = moment(alert.start);
+  const alertEnd = alert.fields[ALERT_END] ? moment(alert.fields[ALERT_END]) : undefined;
+
+  const chartStartToAlertStart = (alert.start - dataTimeRange.from.getTime()) / 1000;
+
+  // Here we check if the available time range before the alert start is long enough
+  // to consider the long window lookback. We consider 3x the long window to be good enough
+  // to cover both the look back within the deviation and the baseline time range.
+  // If the available time range is shorter we fall back to the short window.
+  const lookbackDurationAsSeconds =
+    longWindowLookbackDurationAsSeconds * 3 < chartStartToAlertStart
```

Comment:
By looking at `dataTimeRange` and `chartStartToAlertStart`, it looks like `longWindowLookbackDurationAsSeconds * 3 < chartStartToAlertStart` will never be true. The `chartStartToAlertStart` is going to be same as long time window duration everytime. 
@walterra Maybe I am miscalculating or missing something? Were you able to set `longWindowLookbackDurationAsSeconds * 3` as `lookbackDurationAsSeconds` in your tests?

I tested with "LOW" slo burn rate alert. The time range for log rate analysis is 72 hours. But we use last 72 hours of data to check if alert will be triggered or not. So for analysis we might want to set baseline range before 72 hours (maybe from -144 hours to -72 hours).

<img width="1199" alt="Screenshot 2024-04-16 at 15 53 38" src="https://github.com/elastic/kibana/assets/69037875/7d363d99-f2d2-4002-afe3-dcc46f10d209">

Reply:
@benakansara Good point! We will follow up in a separate PR cc @walterra